test/routes/api/tasks/tasks.test.ts

```diff
@@ -457,182 +469,291 @@ describe('Tasks api (logged user only)', () => {
+        app = await build()
 
-      const taskFilename = encodeURIComponent(`${taskId}_${filename}`)
-      const res = await app.injectWithLogin('basic', {
-        method: 'GET',
-        url: `/api/tasks/${taskFilename}/image`
+        await app.knex<Task>('tasks').where({ id: taskId }).update({ filename: null })
+
+        const files = fs.readdirSync(uploadDirTask)
+        files.forEach((file) => {
+          const filePath = path.join(uploadDirTask, file)
+          fs.rmSync(filePath, { recursive: true })
+        })
+
+        await app.close()
       })
 
-      assert.strictEqual(res.statusCode, 200)
-      assert.strictEqual(res.headers['content-type'], 'image/png')
+      beforeEach(async () => {
+        app = await build()
+        await uploadImageForTask(app, taskId, testImagePath, uploadDirTask)
+        await app.close()
+      })
 
-      const originalFile = fs.readFileSync(testImagePath)
+      after(async () => {
+        const files = fs.readdirSync(uploadDirTask)
+        files.forEach((file) => {
+          const filePath = path.join(uploadDirTask, file)
+          fs.rmSync(filePath, { recursive: true })
+        })
+      })
 
-      assert.deepStrictEqual(originalFile, res.rawPayload)
-    })
+      it('should remove an existing image for a task', async (t) => {
+        app = await build(t)
 
-    it('should return 404 error for non-existant filename', async (t) => {
-      app = await build(t)
+        const taskFilename = encodeURIComponent(`${taskId}_${filename}`)
+        const resDelete = await app.injectWithLogin('basic', {
+          method: 'DELETE',
+          url: `/api/tasks/${taskFilename}/image`
+        })
 
-      const res = await app.injectWithLogin('basic', {
-        method: 'GET',
-        url: '/api/tasks/non-existant/image'
+        assert.strictEqual(resDelete.statusCode, 204)
+
+        const files = fs.readdirSync(uploadDirTask)
+        assert.strictEqual(files.length, 0)
       })
 
-      assert.strictEqual(res.statusCode, 404)
-      const { message } = JSON.parse(res.payload)
-      assert.strictEqual(message, 'No task has filename "non-existant"')
+      it('should return 404 for non-existant task with filename for deletion', async (t) => {
+        app = await build(t)
+
+        const res = await app.injectWithLogin('basic', {
+          method: 'DELETE',
+          url: '/api/tasks/non-existant/image'
+        })
+
+        assert.strictEqual(res.statusCode, 404)
+        const { message } = JSON.parse(res.payload)
+        assert.strictEqual(message, 'No task has filename "non-existant"')
+      })
+
+      it('should return 404 for non-existant file in upload dir', async (t) => {
+        const app = await build(t)
+
+        await createTask(app, {
+          name: 'Task with image',
+          author_id: 1,
+          status: TaskStatusEnum.New,
+          filename: 'does_not_exist.png'
+        })
+
+        const resDelete = await app.injectWithLogin('basic', {
```

Comment:
```suggestion
        const res = await app.injectWithLogin('basic', {
```

`res` is enough context to understand the code.